src/overlays/actors/ovl_En_Kgy/z_en_kgy.c

```diff
@@ -41,105 +63,1132 @@ const ActorInit En_Kgy_InitVars = {
     (ActorFunc)EnKgy_Draw,
 };
 
-#endif
+void EnKgy_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnKgy* this = THIS;
+    s16 cs;
+    s32 i;
+
+    Actor_SetScale(&this->actor, 0.01f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_0600F910, &D_06004B98, this->jointTable, this->morphTable, 23);
+    this->unk_2D2 = -1;
+    this->unk_29C = 0;
+    this->unk_2E4 = 0;
+    this->unk_2E2 = -1;
+    this->unk_2A0 = func_80B40BFC(globalCtx);
+    this->unk_2A4 = func_80B40C38(globalCtx);
+    Actor_UnsetSwitchFlag(globalCtx, ENKGY_PARAMS_FE00(&this->actor) + 1);
+    if (Flags_GetSwitch(globalCtx, ENKGY_PARAMS_FE00(&this->actor)) || (gSaveContext.weekEventReg[33] & 0x80)) {
+        Actor_SetSwitchFlag(globalCtx, ENKGY_PARAMS_FE00(&this->actor) + 1);
+        globalCtx->envCtx.unk_C3 = 1;
+        gSaveContext.weekEventReg[21] |= 1;
+        if (!func_80B40D64(globalCtx)) {
+            func_80B40B38(this, 4, 0, 0);
+            this->actionFunc = func_80B425A0;
+            this->actor.textId = 0xC35;
+        } else if (!func_80B40DB4(globalCtx)) {
+            func_80B40B38(this, 6, 2, 0);
+            this->actionFunc = func_80B419B0;
+            this->actor.textId = 0xC4E;
+            this->unk_29C |= 1;
+            this->unk_2EA = 3;
+        } else {
+            func_80B40B38(this, 4, 0, 0);
+            this->actionFunc = func_80B42714;
+            this->actor.textId = 0xC50;
+        }
+    } else {
+        if (gSaveContext.weekEventReg[20] & 0x80) {
+            func_80B40B38(this, 4, 0, 0);
+        } else {
+            func_80B40B38(this, 0, 0, 0);
+        }
+        this->actionFunc = func_80B42D28;
+    }
+
+    cs = this->actor.cutscene;
+    for (i = 0; i < 6; i++) {
+        this->unk_2D4[i] = cs;
+        if (cs != -1) {
+            this->actor.cutscene = cs;
+            cs = ActorCutscene_GetAdditionalCutscene(this->actor.cutscene);
+        }
+    }
+
+    Lights_PointNoGlowSetInfo(&this->lightInfo, this->actor.world.pos.x, this->actor.world.pos.y,
+                              this->actor.world.pos.z, 0xFF, 0x40, 0x40, -1);
+    this->lightNode = LightContext_InsertLight(globalCtx, &globalCtx->lightCtx, &this->lightInfo);
+    this->unk_300 = -1;
+    this->actor.flags &= ~1;
+}
+
+void EnKgy_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnKgy* this = THIS;
+
+    LightContext_RemoveLight(globalCtx, &globalCtx->lightCtx, this->lightNode);
+}
+
+void func_80B40B38(EnKgy* this, s16 arg1, u8 arg2, f32 arg3) {
```

Comment:
```suggestion
void EnKgy_ChangeAnim(EnKgy* this, s16 animIndex, u8 mode, f32 transitionRate) {
```
perhaps? At least change the arg names to match `SkelAnime_ChangeAnim` since this is mostly just a passthough to call that.